### PR TITLE
Restructure files

### PR DESCRIPTION
**Pull request by AJenbo**
Combine the 3 Source{*} folders into a single one.

Files SourceX/S/* becomes SourceX/S/utils or SourceX/S/storm, folders are moved as is.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -666,18 +652,17 @@ if(NOT CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
...
   target_compile_options(${BIN_TARGET} PRIVATE -Wall -Wextra -Wno-unused-parameter)
 
   # For ARM and other default unsigned char platforms
-  target_compile_options(devilution PRIVATE -fsigned-char)
   target_compile_options(${BIN_TARGET} PRIVATE -fsigned-char)
 endif()
 
 if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
-  target_compile_options(devilution PRIVATE "/W0")
+  target_compile_options(${BIN_TARGET} PRIVATE "/W0")
```
Comment (julealgon) [inline CMakeLists.txt:665]: _Really_  hard to review this file...

File: Source/miniwin/miniwin.h [hunk 2]
```diff
@@ -2,8 +2,22 @@
...
+#include <ctype.h>
+#include <math.h>
+#include "utils/stdarg_compat.h"
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+#include "storm/storm_full.h"
```
Comment (julealgon) [inline Source/miniwin/miniwin.h:14]: I don't understand this part. How could moving the files cause you to have to add all these headers?

File: Source/miniwin/miniwin.h [hunk 3]
```diff
@@ -2,8 +2,22 @@
...
+
 namespace devilution {
 
+#ifndef MAX_PATH
+#define MAX_PATH 260
+#endif
```
Comment (julealgon) [inline Source/miniwin/miniwin.h:20]: ?

File: Source/storm/storm.cpp [hunk 4]
```diff
@@ -15,16 +16,16 @@
...
-#include "stubs.h"
+#include "utils/display.h"
+#include "utils/paths.h"
+#include "utils/sdl_compat.h"
+#include "utils/stubs.h"
 
 namespace devilution {
 
-DWORD nLastError = 0;
+uint32_t nLastError = 0;
```
Comment (julealgon) [inline Source/storm/storm.cpp:28]: Unrelated?

**Review by julealgon — COMMENTED** [review]
Can't really approve this myself as it touches on some things I do not fully understand, My approval would mean nothing.

File: Source/miniwin/miniwin.h (hunk 2 above)
Comment (AJenbo) [reply Source/miniwin/miniwin.h:14]: misc.h was combined with miniwin.h. minwin.h used to be a all.h for the minimal implementation of the Windows API used by Diablo, but at this point only misc.h was left and so one was never used without the other and made better sense as a combined file. This also makes it more clear where we need to clean things up in order to get rid of the last parts of miniwin.

File: Source/storm/storm.cpp (hunk 4 above)
Comment (AJenbo) [reply Source/storm/storm.cpp:28]: Much easier than to try and deal with the nested relation between storm.h and minwin.h. It was either this, have it not compile, or add a hack.

**Comment by galaxyhaxz** [comment]
Great work in unifying the source tree Anders. I think this will help make it a lot more clear and organized now. Diablo's original design of having 100 files in one folder just isn't good design by modern standards. Then again, non of it was :P

File: CMakeLists.txt (hunk 1 above)
Comment (glebm) [reply CMakeLists.txt:665]: This is OK, now there is just one target

**Review by glebm — APPROVED** [review]

File: Source/miniwin/miniwin.h (hunk 3 above)
Comment (AJenbo) [reply Source/miniwin/miniwin.h:20]: The max length of a file path as defined in WinAPI on Windows 95. This may be already defined on Windows platforms and should only be conditionally defined.

Again this comes from the merging of `SourceS/miniwin/misc.h` and `SourceS/miniwin.h`.

Hope that answers you questionmark.

**Comment by AJenbo** [comment]
> Great work in unifying the source tree Anders. I think this will help make it a lot more clear and organized now. Diablo's original design of having 100 files in one folder just isn't good design by modern standards. Then again, non of it was :P

Yeah I'm hoping it will lead to less confusion for new developers looking at the code, it should also make it easier to analyse and run various tools on it :)

File: CMakeLists.txt (hunk 1 above)
Comment (julealgon) [reply CMakeLists.txt:665]: I wasn't specifically talking about this particular change, but the file as a whole. Should've added a file-level comment, sorry for the confusion.

File: Source/storm/storm.cpp (hunk 4 above)
Comment (AJenbo) [reply Source/storm/storm.cpp:28]: ... and it's gone.

https://github.com/diasurgical/devilutionX/commit/e1278cba7b611e6ba130ccbebcff40746e73f65a#diff-7501f2d1b1864ababb342df86d326f82d9f92f4bb34b56b8ed91ac75c87b1a0a
